hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutAllBackupOperation.java

```diff
@@ -52,11 +54,17 @@ public PutAllBackupOperation() {
     public void run() {
         int partitionId = getPartitionId();
         recordStore = mapService.getMapServiceContext().getRecordStore(partitionId, name);
+        boolean wanEnabled = mapContainer.getWanReplicationPublisher() != null && mapContainer.getWanMergePolicy() != null;
```

Comment:
why don't you add a method 'isWanReplicationEnabled' or something to the MapContainer?